Framework/PythonInterface/mantid/FunctionWrapper.py

```diff
@@ -0,0 +1,480 @@
+      
+          :param name: name that appears in the []
+          :param value: new value of this item
+      """
+      if type(name) == type('string') and self.fun.hasAttribute(name):
+         self.fun.setAttributeValue(name, value)
+      else:
+         self.fun.setParameter(name, value)
+      
+  def __str__ (self):
+      """ Return string giving contents of function.
+          Used in unit tests.
+      """
+      return self.fun.__str__()
+                 
+  def __add__ (self, other):
+      """ Implement + operator for composite function
+      
+          :param other: functionWrapper to be added to self
+      """
+      sum = CompositeFunctionWrapper(self, other)
+      if sum.pureAddition:
+         sum = sum.flatten()
+      return sum     
+        
+  def __mul__ (self, other):
+      """ Implement * operator for product function
+      
+          :param other: functionWrapper to multiply self by
+      """
+      prod = ProductFunctionWrapper(self, other)
+      if prod.pureMultiplication:
+         prod = prod.flatten()
+      return prod
+        
+  def tie (self, *args, **kwargs):
+    """ Add ties.
+    """
+    for a in args:
+        if isinstance(a, dict):
+          for key in a:
+            self.fun.tie(key, str(a[key]))
+
+    for key in kwargs:
+        self.fun.tie(key, str(kwargs[key]))
+        
+  def fix(self, name):
+      """ Fix a parameter.
+      
+          :param name: name of parameter to be fixed
+      """
+      self.fun.fixParameter(name)
+      
+  def fixAllParameters(self):
+       """ Fix all parameters.
+       """
+       for i in range(0, self.fun.numParams()):
+          self.fix(self.getParameterName(i))
+      
+  def untie(self, name):
+      """ Remove tie from parameter.
+      
+          :param name: name of parameter to be untied
+      """
+      self.fun.removeTie(name)
+      
+  def untieAllParameters(self):
+      """ Remove ties from all parameters.
+      """
+      for i in range(0, self.fun.numParams()):  
+          self.fun.removeTie(self.getParameterName(i))  
+
+          
+  def constrain(self, expressions):
+      """ Add constraints
+      
+          :param expressions: string of tie expressions      
+      """
+      self.fun.addConstraints( expressions )
+      
+  def unconstrain(self, name):
+      """ Remove constraints from a parameter
+      
+          :param name: name of parameter to be unconstrained
+      """
+      self.fun.removeConstraint(name)
+           
+  def free(self, name):
+      """ Free a parameter from tie or constraint
+      
+          :param name: name of parameter to be freed
+      """
+      self.fun.removeTie(name)
+      self.fun.removeConstraint(name)
+      
+  def getParameterName(self, index):
+      """ Get the name of the parameter of given index
+      
+          :param index: index of parameter
+      """
+      return self.fun.getParamName(index)
+      
+  def getFunction(self):
```

Comment:
@KarlPalmen `function` is the property I wrote above in this thread in order to replace method `getFunction()`. Getters and setters in a Python API are replaced with [property methods](https://stackoverflow.com/questions/17330160/how-does-the-property-decorator-work) because it reads more pythonic: